src/coreclr/jit/importer.cpp

```diff
@@ -4350,7 +4350,7 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                     GenTree*   op1 = impPopStack().val;
                     genTreeOps op  = (ni == NI_System_Threading_Interlocked_Or) ? GT_XORR : GT_XAND;
                     retNode        = gtNewOperNode(op, genActualType(callType), op1, op2);
-                    retNode->gtFlags |= GTF_GLOB_REF | GTF_ASG;
+                    retNode->gtFlags |= GTF_GLOB_REF | GTF_ASG | GTF_EXCEPT;
```

Comment:
Same here: we should we only add `GTF_EXCEPT` if `fgAddrCouldBeNull(op1)` is `true`.